### PR TITLE
Added enumerable

### PR DESCRIPTION
**Pull request by shayan1234554321**
## Added the following

- Create my_enumerable.rb file with the methods (All?, Any? and filter)
- Create MyList class
- Test methods

I am working alone, screenshot attached
Screenshot 2023-05-13 004858

### CODE REVIEW

File: main.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+class CustomList
+  include MyEnumerable
+  def initialize(*items)
+    @items = items
+  end
+
+  def iterate(&block)
+    @items.each { |item| block.call(item) }
+  end
+end
```
Comment (BenMukebo) [inline main.rb]: - [x] According to the project requirements, you should create a class `MyList` that has an instance variable @list. Kindly rename the CustomList class name to `MyList`  as requested by the project requirements. 👇🏻 

- [x] Same here you should implement  In MyList class a method `#each` that yields successive members of @list and uses the MyEnumerable module. Kindly rename your iterate method name. 

image

File: my_enumerable.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# Create MyEnumerable module
+module MyEnumerable
+  def my_all?
+    each do |element|
+      return false unless yield(element)
+    end
+    true
+  end
+
```
Comment (BenMukebo) [inline my_enumerable.rb]: - [x] Here is the result I get when I verify your solution: 👇🏻 Kindly fix this. I'l recommand you add instruction on how to run your code into the README file.

image

**Review by BenMukebo — CHANGES_REQUESTED** [review]
### Hi @shayan1234554321,

You did a great job so far 👏🏼 👏🏼
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

### To highlight 💪🏻
- Gitflow was correctly used.  ✔️   
 👍  Good job!


## STATUS: CHANGES REQUESTED  ♻️ 
_Please Check the comments under the review and make all required changes_


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by zainsadaqat — APPROVED** [review]
# Pull Request Approved ✅ 🎊 🎉 
## Hello @shayan1234554321 👋 

## Things to Appreciate 
- [ ] Correct Flow ✅ 
- [ ] Clean Code 👍 
- [ ] No Linter Errors 🔥 
- [ ] Descriptive Pull Request and Summary 👌 

### Your project is complete!  There is nothing else to say other than it's time to merge it.

# Congratulations!!! 🎉 🥳 👏 🙌 🎊 ☑️ 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**